clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -56,7 +56,14 @@ using namespace CodeGen;
 // Experiment to make sanitizers easier to debug
 static llvm::cl::opt<bool> ClSanitizeDebugDeoptimization(
     "ubsan-unique-traps", llvm::cl::Optional,
-    llvm::cl::desc("Deoptimize traps for UBSAN so there is 1 trap per check"),
+    llvm::cl::desc("Deoptimize traps for UBSAN so there is 1 trap per check."),
+    llvm::cl::init(false));
+
+// TODO: Introduce frontend options to enabled per sanitizers, similar to
+// `fsanitize-trap`.
+static llvm::cl::opt<bool> ClSanitizeExpHot(
+    "ubsan-exp-hot", llvm::cl::Optional,
+    llvm::cl::desc("Pass UBSAN checks if `llvm.experimental.hot()` is true."),
```

Comment:
I'm not sure how best to do it, but it might be useful to have a remark that fires when one of these is optimized out.

Reply:
Sorry, messed up the spr stack. fixing

Reply:
@jroelofs  It should be in a different one, I'll add you into the review

Reply:
I guess #84858

Reply:
moved here: https://github.com/llvm/llvm-project/pull/84858/files#r1520603139